uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenPaymentPresenter.kt

```diff
@@ -34,20 +44,32 @@ class AdyenPaymentPresenter(view: PaymentMVP.View,
         attachView(view)
     }
 
-    override fun sdkInit(price: QuotePrice?) {
-        this.price = price
-        paymentsService.getAdyenPublicKey().execute { result ->
-            when (result) {
-                is Resource.Success -> {
-                    result.data.let {
-                        adyenKey = it.publicKey
-                        getPaymentMethods()
-                    }
-                }
-                //TODO Change error message
-                is Resource.Failure -> view?.showError(R.string.payment_issue_message)
-            }
+    override fun getDropInConfig(context: Context, sdkToken: String): Any {
+        val cardConfiguration =
+                CardConfiguration.Builder(context, sdkToken)
+                        .setShopperLocale(Locale.getDefault())
+                        .setHolderNameRequire(true)
+                        .build()
+
+        val dropInIntent = Intent(context, AdyenResultActivity::class.java).apply {
+            putExtra(AdyenResultActivity.TYPE_KEY, ComponentType.DROPIN.id)
```

Comment:
The `ComponentType` can be removed or extracted to a separate class (instead of being part of AdyenPaymentView ).

Reply:
Moved